src/Servers/Kestrel/stress/Program.cs

```diff
@@ -112,6 +112,54 @@ void ValidateContent(string expectedContent, string actualContent)
             }
         }
 
+        Func<ClientContext, Task> TestAbort(string path)
+        {
+            return async ctx =>
+            {
+                Version httpVersion = ctx.GetRandomVersion(httpVersions);
+                try
+                {
+                    using (var req = new HttpRequestMessage(HttpMethod.Get, serverUri + path) { Version = httpVersion })
+                    {
+                        await ctx.HttpClient.SendAsync(req);
+                    }
+                    throw new Exception("Completed unexpectedly");
+                }
+                catch (Exception e)
+                {
+                    if (e is HttpRequestException hre && hre.InnerException is IOException)
+                    {
+                        e = hre.InnerException;
+                    }
+
+                    if (e is IOException ioe)
+                    {
+                        if (httpVersion < HttpVersion.Version20)
+                        {
+                            return;
+                        }
+
+                        string name = e.InnerException?.GetType().Name;
+                        switch (name)
+                        {
+                            case "Http2ProtocolException":
+                            case "Http2ConnectionException":
+                            case "Http2StreamException":
+                                if (e.InnerException.Message.Contains("INTERNAL_ERROR") || e.InnerException.Message.Contains("CANCEL"))
```

Comment:
```suggestion
                                // This can be improved when https://github.com/dotnet/runtime/issues/43239 is available.
                                if (e.InnerException.Message.Contains("INTERNAL_ERROR") || e.InnerException.Message.Contains("CANCEL"))
```